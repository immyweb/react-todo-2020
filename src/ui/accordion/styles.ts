import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
