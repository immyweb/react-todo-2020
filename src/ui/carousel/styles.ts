import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { BtnNoStyle } from '../../styles/globals';
import { Direction } from './types';

export const CarouselContainer = glamorous.section({
  label: 'carousel-container',
  paddingBottom: 33,
  marginBottom: 25,
  position: 'relative',
  width: '100%',
});

export const Slide = glamorous.div({
  label: 'carousel-slide',
  cursor: 'pointer',
  padding: '0 10px',
  ':focus': {
    outline: 'none',
  },
});

export const SlideLink = glamorous.a({
  label: 'carousel-link',
  display: 'inline-block',
  width: '100%',
  height: '100%',
  objectFit: 'contain',
  textDecoration: 'none',
});

export const SlideImg = glamorous.img<{ borderColour: string }>(
  {
    label: 'carousel-image',
    width: '100%',
    height: '100%',
    objectFit: 'contain',
  },
  ({ borderColour }) => ({
    borderBottom: `3px solid ${borderColour}`,
  }),
);

export const SlideLabel = glamorous.p<{ txtColour: string }>(
  {
    label: 'carousel-label',
    textAlign: 'center',
    fontFamily: fonts.$theSunHeavyNarrow,
    fontSize: 18,
    paddingTop: 8,
  },
  ({ txtColour }) => ({
    color: txtColour,
  }),
);

export const SlideNav = glamorous(BtnNoStyle)<{ direction: Direction }>(
  {
    label: 'carousel-nav',
    position: 'absolute',
    zIndex: 3,
    bottom: -47,
    width: 48,
    height: 48,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ direction }) => ({
    left: direction === Direction.PREV ? 25 : 'auto',
    right: direction === Direction.NEXT ? 25 : 'auto',
  }),
);

// Pagination

export const PaginationContainer = glamorous.div({
  label: 'pagination-container',
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
  width: '100%',
  height: 20,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const Dot = glamorous.span<{ colour?: string; label?: string }>(
  {
    display: 'inline-block',
    borderRadius: '50%',
    margin: '0 5px',
  },
  ({ colour, label }) => ({
    label,
    background: colour ? colour : colours.$news,
  }),
);
