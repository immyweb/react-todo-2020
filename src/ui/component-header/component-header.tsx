import * as React from 'react';

import { TitleWrapper, Highlight, Title, SubTitle } from './styles';
import { fonts } from '../../styles/fonts';

import { colours } from '../../styles/colours';
import { ITheme } from '../../types/theme';

import { IAlign } from '../../utils/alignment-utils';
import { dangerouslySet } from '../../utils/html-utils';

interface IProps {
  className?: string;
  inline?: boolean;
  highlight?: string;
  title?: string;
  subTitle?: string;
  theme?: ITheme;
  alignment?: IAlign;
  background?: boolean;
}

const defaultProps = {
  inline: false,
  theme: {
    primary: colours.$main,
    secondary: colours.$mainSecondary,
  },
  alignment: {
    x: 'left',
  },
  background: false,
};

export const ComponentHeader: React.SFC<IProps> = ({
  className,
  inline,
  highlight,
  title,
  subTitle,
  theme,
  alignment,
  background,
}) => {
  const _inline = inline ? inline : defaultProps.inline;
  const _theme = theme ? theme : defaultProps.theme;
  const _alignment = alignment ? alignment : defaultProps.alignment;
  const _background = background ? background : defaultProps.background;

  const returnText = () => {
    if (background) {
      return (
        <Title
          fontName={fonts.$theSunHeavyCondensed}
          inline={_inline}
          alignment={_alignment}
          background={_background}
        >
          {title}
        </Title>
      );
    }

    return (
      <div>
        {highlight && (
          <Highlight
            fontName={fonts.$theSunHeavyCondensed}
            fontColour={_theme.primary}
            inline={_inline}
            alignment={_alignment}
            background={_background}
          >
            {highlight}
          </Highlight>
        )}
        <Title
          fontName={fonts.$theSunHeavyCondensed}
          inline={_inline}
          alignment={_alignment}
          background={_background}
        >
          {title}
        </Title>
      </div>
    );
  };

  return (
    <div className={className}>
      <TitleWrapper
        theme={theme}
        inline={_inline}
        alignment={_alignment}
        background={_background}
        className={`${className}-inner`}
      >
        {returnText()}
      </TitleWrapper>
      {subTitle && !background && (
        <SubTitle
          className={`${className}-subtitle`}
          linkColour={_theme.primary}
          alignment={_alignment}
          dangerouslySetInnerHTML={dangerouslySet(`${subTitle}`)}
          hasBackground={_background}
        />
      )}
    </div>
  );
};
