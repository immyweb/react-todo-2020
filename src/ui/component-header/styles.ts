import glamorous from 'glamorous';

import { P, H2 } from '../typography/typography';
import { colours } from '../../styles/colours';

import { mediaQueries } from '../../styles/breakpoints';
import { IAlign } from '../../utils/alignment-utils';

interface IGProps {
  inline?: boolean;
  alignment?: IAlign;
  linkColour?: string;
  background?: boolean;
  hasBackground?: boolean;
}

const styles = {
  fontSizeLarge: 28,
  fontSizeSmall: 24,
  inlineHighlight: 16,
  marginBottomLarge: 12,
  h2LineHeight: 1.14,
  pLineHeight: 1.38,
};

export const TitleWrapper = glamorous.div<IGProps>(
  {
    label: 'title-wrapper',
  },
  ({ alignment, background, theme }: any) => ({
    textAlign: alignment.x,
    background: background ? theme.primary : 'none',
    padding: background ? '13px 20px 13px 15px' : 0,
    borderLeftStyle: background ? 'solid' : 'none',
    borderLeftWidth: '5px',
    borderLeftColor: theme.secondary,
  }),
);

export const Title = glamorous(H2)<IGProps>(
  {
    label: 'title',
    fontSize: styles.fontSizeSmall,
    lineHeight: styles.h2LineHeight,
    [mediaQueries.tabletPortrait]: {
      fontSize: styles.fontSizeLarge,
      lineHeight: styles.h2LineHeight,
    },
  },
  ({ inline, alignment, background }: any) => ({
    display: inline ? 'inline-block' : 'block',
    textTransform: background ? 'unset' : inline ? 'unset' : 'uppercase',
    marginBottom: background ? 0 : styles.marginBottomLarge,
    textAlign: alignment.x,
    color: background ? colours.$white : colours.$offBlack,
  }),
);

export const Highlight = glamorous(Title)<IGProps>(
  {
    label: 'highlight',
    marginBottom: 0,
    fontSize: styles.fontSizeSmall,
    [mediaQueries.tabletPortrait]: {
      fontSize: styles.fontSizeLarge,
      marginBottom: 0,
    },
  },
  ({ inline, alignment, fontColour, background }: any) => ({
    fontSize: inline ? styles.fontSizeSmall : styles.inlineHighlight,
    textTransform: 'uppercase',
    textAlign: alignment.x,
    color: fontColour,
    marginRight: inline ? 4 : 0,
    [mediaQueries.tabletPortrait]: {
      fontSize: inline ? styles.fontSizeLarge : styles.inlineHighlight,
    },
  }),
);

export const SubTitle = glamorous(P)<IGProps>(
  {
    label: 'sub-title',
    lineHeight: styles.pLineHeight,
    textTransform: 'lowercase',
    marginBottom: 20,
    [mediaQueries.tabletPortrait]: {
      lineHeight: styles.pLineHeight,
    },
  },
  ({ alignment, linkColour, hasBackground }: any) => ({
    textAlign: alignment.x,
    marginTop: hasBackground ? 20 : 0,
    '& a': {
      color: linkColour,
      textDecoration: 'none',
    },

    '& a:hover': {
      textDecoration: 'underline',
    },
  }),
);
