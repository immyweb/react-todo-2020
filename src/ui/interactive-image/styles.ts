import glamorous from 'glamorous';

import { mediaQueries } from '../../styles/breakpoints';

export const InteractiveImageContainer = glamorous.section({
  label: 'interactive-image-container',
  width: '100%',
  height: '100%',
});

export const InteractiveImageInner = glamorous.div({
  label: 'interactive-image-inner',
  position: 'relative',
  width: '100%',
  height: '100%',
});

export const Image = glamorous.img({
  label: 'interactive-image',
  position: 'relative',
  width: '100%',
});

export const Hotspot = glamorous.div<{
  top: number;
  left: number;
}>(
  {
    label: 'interactive-hotspot',
    position: 'absolute',
    zIndex: 5,
  },
  ({ top, left }) => ({
    top: `${top}%`,
    left: `${left}%`,
    [mediaQueries.tabletPortrait]: {
      top: `${top}%`,
      left: `${left}%`,
    },
  }),
);
