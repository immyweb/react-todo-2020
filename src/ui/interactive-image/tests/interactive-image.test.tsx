import * as React from 'react';
import { shallow } from 'enzyme';
import InteractiveImage from '../';
import { ComponentHeader } from '../../component-header';

import { Image } from '../styles';
import { ArrowPosition } from '../../label/label';

const data = {
  config: {
    headlineTheme: {
      primary: '#4085cb',
      secondary: '#2d5c8c',
    },
    labelTheme: {
      txtColour: '#ffffff',
      bkgndColour: '#000000',
      pulseColour: '#eeeeee',
    },
  },
  content: {
    headline: 'Body parts',
    image:
      'https://www.thesun.co.uk/wp-content/uploads/2018/04/nintchdbpict000399253435.jpg?strip=all&w=513',
    labels: [
      {
        label: {
          text: 'ARMS',
          arrowSide: ArrowPosition.BOTTOM,
          arrowPosition: ArrowPosition.LEFT,
          mobilePosition: {
            top: '30%',
            left: '65%',
          },
          desktopPosition: {
            top: '30%',
            left: '65%',
          },
        },
        modal: {
          title: 'Long arms are a turn on',
          imageTop:
            'https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960',
          copy: [
            'Longer arms make you more attractive, according to the University of New South Wales. Long dimensions in the arms can send a subtle signal that youâ€™re a suitable mate.',
          ],
        },
      },
      {
        label: {
          text: 'BOOBS',
          arrowSide: ArrowPosition.BOTTOM,
          arrowPosition: ArrowPosition.RIGHT,
          mobilePosition: {
            top: '23%',
            left: '23%',
          },
          desktopPosition: {
            top: '30%',
            left: '65%',
          },
        },
        modal: {
          title: 'Bigger boobs mean bigger IQs',
          imageBottom:
            'https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960',
          copy: [
            'Ladies with larger breasts scored an average of 10 points higher on an IQ test as part of a University of Chicago study. This may be because the hormones which make boobs grow play a role in intelligence.',
          ],
        },
      },
    ],
  },
};

function shallowRender(props?: any) {
  const { config, content } = data;
  return shallow(<InteractiveImage config={config} content={content} {...props} />);
}

describe('<InteractiveImage />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the provided image', () => {
    const wrapper = shallowRender();
    expect(wrapper.find(Image).props().src).toEqual(
      'https://www.thesun.co.uk/wp-content/uploads/2018/04/nintchdbpict000399253435.jpg?strip=all&w=513',
    );
    expect(wrapper.find(Image).props().alt).toEqual('Body parts');
  });

  it('renders correct headline and theme', () => {
    const wrapper = shallowRender();
    const header = wrapper.find(ComponentHeader);
    expect(header.props().title).toEqual('Body parts');
    expect(header.props().theme).toEqual({
      primary: '#4085cb',
      secondary: '#2d5c8c',
    });
  });

  it('renders correct DOM element and ID', () => {
    const wrapper = shallowRender();
    const dom = wrapper.find('#interactiveImageModal');
    expect(dom.length).toEqual(1);
  });
});
