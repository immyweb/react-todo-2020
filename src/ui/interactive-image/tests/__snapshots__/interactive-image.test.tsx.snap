// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InteractiveImage /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    config={
      Object {
        "headlineTheme": Object {
          "primary": "#4085cb",
          "secondary": "#2d5c8c",
        },
        "labelTheme": Object {
          "bkgndColour": "#000000",
          "pulseColour": "#eeeeee",
          "txtColour": "#ffffff",
        },
      }
    }
    content={
      Object {
        "headline": "Body parts",
        "image": "https://www.thesun.co.uk/wp-content/uploads/2018/04/nintchdbpict000399253435.jpg?strip=all&w=513",
        "labels": Array [
          Object {
            "label": Object {
              "arrowPosition": "LEFT",
              "arrowSide": "BOTTOM",
              "desktopPosition": Object {
                "left": "65%",
                "top": "30%",
              },
              "mobilePosition": Object {
                "left": "65%",
                "top": "30%",
              },
              "text": "ARMS",
            },
            "modal": Object {
              "copy": Array [
                "Longer arms make you more attractive, according to the University of New South Wales. Long dimensions in the arms can send a subtle signal that you’re a suitable mate.",
              ],
              "imageTop": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
              "title": "Long arms are a turn on",
            },
          },
          Object {
            "label": Object {
              "arrowPosition": "RIGHT",
              "arrowSide": "BOTTOM",
              "desktopPosition": Object {
                "left": "65%",
                "top": "30%",
              },
              "mobilePosition": Object {
                "left": "23%",
                "top": "23%",
              },
              "text": "BOOBS",
            },
            "modal": Object {
              "copy": Array [
                "Ladies with larger breasts scored an average of 10 points higher on an IQ test as part of a University of Chicago study. This may be because the hormones which make boobs grow play a role in intelligence.",
              ],
              "imageBottom": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
              "title": "Bigger boobs mean bigger IQs",
            },
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown
          background={true}
          className="interactive-image-header"
          theme={
            Object {
              "primary": "#4085cb",
              "secondary": "#2d5c8c",
            }
          }
          title="Body parts"
        />,
        <div
          id="interactiveImageModal"
        />,
        <glamorous(div)>
          <Unknown
            hotSpotBkgndColour="#000000"
            hotSpotTxtColour="#ffffff"
            hotspots={
              Array [
                Object {
                  "label": Object {
                    "arrowPosition": "LEFT",
                    "arrowSide": "BOTTOM",
                    "desktopPosition": Object {
                      "left": "65%",
                      "top": "30%",
                    },
                    "mobilePosition": Object {
                      "left": "65%",
                      "top": "30%",
                    },
                    "text": "ARMS",
                  },
                  "modal": Object {
                    "copy": Array [
                      "Longer arms make you more attractive, according to the University of New South Wales. Long dimensions in the arms can send a subtle signal that you’re a suitable mate.",
                    ],
                    "imageTop": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
                    "title": "Long arms are a turn on",
                  },
                },
                Object {
                  "label": Object {
                    "arrowPosition": "RIGHT",
                    "arrowSide": "BOTTOM",
                    "desktopPosition": Object {
                      "left": "65%",
                      "top": "30%",
                    },
                    "mobilePosition": Object {
                      "left": "23%",
                      "top": "23%",
                    },
                    "text": "BOOBS",
                  },
                  "modal": Object {
                    "copy": Array [
                      "Ladies with larger breasts scored an average of 10 points higher on an IQ test as part of a University of Chicago study. This may be because the hormones which make boobs grow play a role in intelligence.",
                    ],
                    "imageBottom": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
                    "title": "Bigger boobs mean bigger IQs",
                  },
                },
              ]
            }
            modalRef={[Function]}
            pulseColour="#eeeeee"
          />
          <glamorous(img)
            alt="Body parts"
            className="interactive-image-img"
            src="https://www.thesun.co.uk/wp-content/uploads/2018/04/nintchdbpict000399253435.jpg?strip=all&w=513"
          />
        </glamorous(div)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "background": true,
          "className": "interactive-image-header",
          "theme": Object {
            "primary": "#4085cb",
            "secondary": "#2d5c8c",
          },
          "title": "Body parts",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "interactiveImageModal",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Unknown
              hotSpotBkgndColour="#000000"
              hotSpotTxtColour="#ffffff"
              hotspots={
                Array [
                  Object {
                    "label": Object {
                      "arrowPosition": "LEFT",
                      "arrowSide": "BOTTOM",
                      "desktopPosition": Object {
                        "left": "65%",
                        "top": "30%",
                      },
                      "mobilePosition": Object {
                        "left": "65%",
                        "top": "30%",
                      },
                      "text": "ARMS",
                    },
                    "modal": Object {
                      "copy": Array [
                        "Longer arms make you more attractive, according to the University of New South Wales. Long dimensions in the arms can send a subtle signal that you’re a suitable mate.",
                      ],
                      "imageTop": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
                      "title": "Long arms are a turn on",
                    },
                  },
                  Object {
                    "label": Object {
                      "arrowPosition": "RIGHT",
                      "arrowSide": "BOTTOM",
                      "desktopPosition": Object {
                        "left": "65%",
                        "top": "30%",
                      },
                      "mobilePosition": Object {
                        "left": "23%",
                        "top": "23%",
                      },
                      "text": "BOOBS",
                    },
                    "modal": Object {
                      "copy": Array [
                        "Ladies with larger breasts scored an average of 10 points higher on an IQ test as part of a University of Chicago study. This may be because the hormones which make boobs grow play a role in intelligence.",
                      ],
                      "imageBottom": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
                      "title": "Bigger boobs mean bigger IQs",
                    },
                  },
                ]
              }
              modalRef={[Function]}
              pulseColour="#eeeeee"
            />,
            <glamorous(img)
              alt="Body parts"
              className="interactive-image-img"
              src="https://www.thesun.co.uk/wp-content/uploads/2018/04/nintchdbpict000399253435.jpg?strip=all&w=513"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "hotSpotBkgndColour": "#000000",
              "hotSpotTxtColour": "#ffffff",
              "hotspots": Array [
                Object {
                  "label": Object {
                    "arrowPosition": "LEFT",
                    "arrowSide": "BOTTOM",
                    "desktopPosition": Object {
                      "left": "65%",
                      "top": "30%",
                    },
                    "mobilePosition": Object {
                      "left": "65%",
                      "top": "30%",
                    },
                    "text": "ARMS",
                  },
                  "modal": Object {
                    "copy": Array [
                      "Longer arms make you more attractive, according to the University of New South Wales. Long dimensions in the arms can send a subtle signal that you’re a suitable mate.",
                    ],
                    "imageTop": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
                    "title": "Long arms are a turn on",
                  },
                },
                Object {
                  "label": Object {
                    "arrowPosition": "RIGHT",
                    "arrowSide": "BOTTOM",
                    "desktopPosition": Object {
                      "left": "65%",
                      "top": "30%",
                    },
                    "mobilePosition": Object {
                      "left": "23%",
                      "top": "23%",
                    },
                    "text": "BOOBS",
                  },
                  "modal": Object {
                    "copy": Array [
                      "Ladies with larger breasts scored an average of 10 points higher on an IQ test as part of a University of Chicago study. This may be because the hormones which make boobs grow play a role in intelligence.",
                    ],
                    "imageBottom": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
                    "title": "Bigger boobs mean bigger IQs",
                  },
                },
              ],
              "id": undefined,
              "modalRef": [Function],
              "pulseColour": "#eeeeee",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "Body parts",
              "className": "interactive-image-img",
              "src": "https://www.thesun.co.uk/wp-content/uploads/2018/04/nintchdbpict000399253435.jpg?strip=all&w=513",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown
            background={true}
            className="interactive-image-header"
            theme={
              Object {
                "primary": "#4085cb",
                "secondary": "#2d5c8c",
              }
            }
            title="Body parts"
          />,
          <div
            id="interactiveImageModal"
          />,
          <glamorous(div)>
            <Unknown
              hotSpotBkgndColour="#000000"
              hotSpotTxtColour="#ffffff"
              hotspots={
                Array [
                  Object {
                    "label": Object {
                      "arrowPosition": "LEFT",
                      "arrowSide": "BOTTOM",
                      "desktopPosition": Object {
                        "left": "65%",
                        "top": "30%",
                      },
                      "mobilePosition": Object {
                        "left": "65%",
                        "top": "30%",
                      },
                      "text": "ARMS",
                    },
                    "modal": Object {
                      "copy": Array [
                        "Longer arms make you more attractive, according to the University of New South Wales. Long dimensions in the arms can send a subtle signal that you’re a suitable mate.",
                      ],
                      "imageTop": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
                      "title": "Long arms are a turn on",
                    },
                  },
                  Object {
                    "label": Object {
                      "arrowPosition": "RIGHT",
                      "arrowSide": "BOTTOM",
                      "desktopPosition": Object {
                        "left": "65%",
                        "top": "30%",
                      },
                      "mobilePosition": Object {
                        "left": "23%",
                        "top": "23%",
                      },
                      "text": "BOOBS",
                    },
                    "modal": Object {
                      "copy": Array [
                        "Ladies with larger breasts scored an average of 10 points higher on an IQ test as part of a University of Chicago study. This may be because the hormones which make boobs grow play a role in intelligence.",
                      ],
                      "imageBottom": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
                      "title": "Bigger boobs mean bigger IQs",
                    },
                  },
                ]
              }
              modalRef={[Function]}
              pulseColour="#eeeeee"
            />
            <glamorous(img)
              alt="Body parts"
              className="interactive-image-img"
              src="https://www.thesun.co.uk/wp-content/uploads/2018/04/nintchdbpict000399253435.jpg?strip=all&w=513"
            />
          </glamorous(div)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "background": true,
            "className": "interactive-image-header",
            "theme": Object {
              "primary": "#4085cb",
              "secondary": "#2d5c8c",
            },
            "title": "Body parts",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "interactiveImageModal",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Unknown
                hotSpotBkgndColour="#000000"
                hotSpotTxtColour="#ffffff"
                hotspots={
                  Array [
                    Object {
                      "label": Object {
                        "arrowPosition": "LEFT",
                        "arrowSide": "BOTTOM",
                        "desktopPosition": Object {
                          "left": "65%",
                          "top": "30%",
                        },
                        "mobilePosition": Object {
                          "left": "65%",
                          "top": "30%",
                        },
                        "text": "ARMS",
                      },
                      "modal": Object {
                        "copy": Array [
                          "Longer arms make you more attractive, according to the University of New South Wales. Long dimensions in the arms can send a subtle signal that you’re a suitable mate.",
                        ],
                        "imageTop": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
                        "title": "Long arms are a turn on",
                      },
                    },
                    Object {
                      "label": Object {
                        "arrowPosition": "RIGHT",
                        "arrowSide": "BOTTOM",
                        "desktopPosition": Object {
                          "left": "65%",
                          "top": "30%",
                        },
                        "mobilePosition": Object {
                          "left": "23%",
                          "top": "23%",
                        },
                        "text": "BOOBS",
                      },
                      "modal": Object {
                        "copy": Array [
                          "Ladies with larger breasts scored an average of 10 points higher on an IQ test as part of a University of Chicago study. This may be because the hormones which make boobs grow play a role in intelligence.",
                        ],
                        "imageBottom": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
                        "title": "Bigger boobs mean bigger IQs",
                      },
                    },
                  ]
                }
                modalRef={[Function]}
                pulseColour="#eeeeee"
              />,
              <glamorous(img)
                alt="Body parts"
                className="interactive-image-img"
                src="https://www.thesun.co.uk/wp-content/uploads/2018/04/nintchdbpict000399253435.jpg?strip=all&w=513"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "hotSpotBkgndColour": "#000000",
                "hotSpotTxtColour": "#ffffff",
                "hotspots": Array [
                  Object {
                    "label": Object {
                      "arrowPosition": "LEFT",
                      "arrowSide": "BOTTOM",
                      "desktopPosition": Object {
                        "left": "65%",
                        "top": "30%",
                      },
                      "mobilePosition": Object {
                        "left": "65%",
                        "top": "30%",
                      },
                      "text": "ARMS",
                    },
                    "modal": Object {
                      "copy": Array [
                        "Longer arms make you more attractive, according to the University of New South Wales. Long dimensions in the arms can send a subtle signal that you’re a suitable mate.",
                      ],
                      "imageTop": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
                      "title": "Long arms are a turn on",
                    },
                  },
                  Object {
                    "label": Object {
                      "arrowPosition": "RIGHT",
                      "arrowSide": "BOTTOM",
                      "desktopPosition": Object {
                        "left": "65%",
                        "top": "30%",
                      },
                      "mobilePosition": Object {
                        "left": "23%",
                        "top": "23%",
                      },
                      "text": "BOOBS",
                    },
                    "modal": Object {
                      "copy": Array [
                        "Ladies with larger breasts scored an average of 10 points higher on an IQ test as part of a University of Chicago study. This may be because the hormones which make boobs grow play a role in intelligence.",
                      ],
                      "imageBottom": "https://www.thesun.co.uk/wp-content/uploads/2018/04/gettyimages-104790256.jpg?strip=all&w=960",
                      "title": "Bigger boobs mean bigger IQs",
                    },
                  },
                ],
                "id": undefined,
                "modalRef": [Function],
                "pulseColour": "#eeeeee",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Body parts",
                "className": "interactive-image-img",
                "src": "https://www.thesun.co.uk/wp-content/uploads/2018/04/nintchdbpict000399253435.jpg?strip=all&w=513",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
